lib/commands/ci.js

```diff
@@ -55,6 +56,28 @@ class CI extends ArboristWorkspaceCmd {
       }),
       removeNodeModules(where),
     ])
+
+    // retrieves inventory of packages from loaded virtual tree (lock file)
+    const virtualInventory = new Map(arb.virtualTree.inventory)
+
+    // build ideal tree step needs to come right after retrieving the virtual
+    // inventory since it's going to erase the previous ref to virtualTree
+    await arb.buildIdealTree()
+
+    // verifies that the packages from the ideal tree will match
+    // the same versions that are present in the virtual tree (lock file)
+    // throws a validation error in case of mismatches
+    const errors = validateLockfile(virtualInventory, arb.idealTree.inventory)
```

Comment:
This is a single-use function for this command only.  Why extract it into a separate file?  Can we leave it in here, and let the tests be true integration tests for CI?

Reply:
it sounds nice but the reality is that the tests we have in place for `npm ci` today are also super unitary tests mocking the world and testing the implementation, we would end up with the same kind of tests but with 10x more pieces floating around / mocked surface.

I would push back for having the refactor of the tests as a whole to be a tackled as part of a different story - if that's what we want.